src/client/signatures/rc_crypto_verifier.rs

```diff
@@ -16,70 +14,41 @@ impl Verification for RcCryptoVerifier {
-        let root_fingerprint_bytes = match digest(&SHA256, &root_pem.contents) {
-            Ok(v) => v.as_ref().to_vec(),
-            Err(e) => {
-                return Err(SignatureError::HashingError(e.to_string()));
+        contentsignature::verify(
+            &message,
+            &signature,
+            &pem_bytes,
+            epoch_seconds,
+            &root_hash,
+            &subject_cn,
+        )
+        .map_err(|err| match err.kind() {
+            RcErrorKind::RootHashFormatError(detail) => {
+                SignatureError::RootHashFormatError(detail.to_string())
             }
-        };
-        if root_fingerprint_bytes != root_hash {
-            return Err(SignatureError::CertificateHasWrongRoot(hex::encode(
-                root_fingerprint_bytes,
-            )));
-        }
-        let now = ASN1Time::from_timestamp(epoch_seconds as i64);
-        for cert in &certs {
-            if !cert.tbs_certificate.validity.is_valid_at(now) {
-                return Err(SignatureError::CertificateExpired);
+            RcErrorKind::PEMFormatError(detail) => {
+                SignatureError::CertificateContentError(detail.to_string())
             }
-        }
-        let leaf_cert = certs.last().unwrap(); // PEM parse fails if len == 0.
-        let leaf_subject = leaf_cert
-            .tbs_certificate
-            .subject
-            .iter_common_name()
-            .next()
-            .and_then(|cn| cn.as_str().ok())
-            .unwrap_or("")
-            .to_string();
-        if leaf_subject != subject_cn {
-            return Err(SignatureError::WrongSignerName(leaf_subject));
-        }
-        let public_key_bytes = leaf_cert
-            .tbs_certificate
-            .subject_pki
-            .subject_public_key
-            .data;
-        let signature_alg = &signature::ECDSA_P384_SHA384;
-        let public_key = signature::UnparsedPublicKey::new(signature_alg, &public_key_bytes);
-        match public_key.verify(&message, &signature) {
-            Ok(_) => Ok(()),
-            Err(err) => Err(SignatureError::MismatchError(err.to_string())),
-        }
+            RcErrorKind::CertificateContentError(detail) => {
+                SignatureError::CertificateContentError(detail.to_string())
+            }
+            RcErrorKind::SignatureContentError(detail) => {
+                SignatureError::BadSignatureContent(detail.to_string())
+            }
+            RcErrorKind::CertificateSubjectError => {
+                SignatureError::InvalidCertificateSubject("".to_string())
+            }
+            RcErrorKind::CertificateIssuerError => {
+                SignatureError::InvalidCertificateIssuer("".to_string())
+            }
+            RcErrorKind::CertificateValidityError => SignatureError::CertificateExpired,
+            RcErrorKind::CertificateChainError(_) => SignatureError::CertificateTrustError,
+            _ => SignatureError::MismatchError(err.to_string()),
+        })
```

Comment:
Personally, I'd do this as a `impl From<RcErrorKind> for SignatureError`. The code would be pretty much the same, just in a different place. This function would then end up more like 

```rust
contentsignature::verify(...)?;
Ok(())
```